src/catalog/catalog.cpp

```diff
@@ -521,254 +904,188 @@ std::unique_ptr<storage::DataTable> Catalog::CreateMetricsCatalog(
     schema = InitializeIndexMetricsSchema().release();
   }
 
-  std::unique_ptr<storage::DataTable> table(
-      storage::TableFactory::GetDataTable(database_id, GetNextOid(), schema,
-          table_name, DEFAULT_TUPLES_PER_TILEGROUP, own_schema, adapt_table, is_catalog));
+  std::unique_ptr<storage::DataTable> table(storage::TableFactory::GetDataTable(
+      database_id, TableCatalog::GetInstance()->GetNextOid(), schema,
+      table_name, DEFAULT_TUPLES_PER_TILEGROUP, own_schema, adapt_table,
+      is_catalog));
 
   return table;
 }
 
-// Initialize tables catalog schema
-std::unique_ptr<catalog::Schema> Catalog::InitializeTablesSchema() {
-  const std::string not_null_constraint_name = "not_null";
-
-  auto id_column = catalog::Column(type::Type::INTEGER,
-                                   type::Type::GetTypeSize(type::Type::INTEGER),
-                                   "table_id", true);
-  id_column.AddConstraint(
-      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
-
-  auto name_column = catalog::Column(type::Type::VARCHAR, max_name_size,
-      "table_name", true);
-  name_column.AddConstraint(
-      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
-
-  auto database_id_column = catalog::Column(type::Type::INTEGER,
-      type::Type::GetTypeSize(type::Type::INTEGER), "database_id", true);
-  database_id_column.AddConstraint(
-      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
-
-  auto database_name_column = catalog::Column(type::Type::VARCHAR,
-      max_name_size, "database_name", true);
-  database_name_column.AddConstraint(
-      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
-
-  std::unique_ptr<catalog::Schema> table_schema(new catalog::Schema( {
-      id_column, name_column, database_id_column, database_name_column }));
-
-  return table_schema;
-}
-
-// Initialize database catalog schema
-std::unique_ptr<catalog::Schema> Catalog::InitializeDatabaseSchema() {
-  const std::string not_null_constraint_name = "not_null";
-
-  auto id_column = catalog::Column(type::Type::INTEGER,
-                                   type::Type::GetTypeSize(type::Type::INTEGER),
-                                   "database_id", true);
-  id_column.AddConstraint(
-      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
-  auto name_column = catalog::Column(type::Type::VARCHAR, max_name_size,
-      "database_name", true);
-  name_column.AddConstraint(
-      catalog::Constraint(ConstraintType::NOTNULL, not_null_constraint_name));
-
-  std::unique_ptr<catalog::Schema> database_schema(new catalog::Schema( {
-      id_column, name_column }));
-  return database_schema;
-}
-
-// Initialize database catalog schema
 std::unique_ptr<catalog::Schema> Catalog::InitializeDatabaseMetricsSchema() {
```

Comment:
[Those functions](https://github.com/AngLi-Leon/peloton/blob/master/src/catalog/catalog.cpp#L862) in catalog.cpp are for metrics, we didn't change anything in those functions and nobody seems to use them right now.